### ISSUE TITLE
plotly_afterplot event doesn't fire on scatterternary plots

### ISSUE THREAD

**Issue by dannycodes** [plotly/plotly.js#1498]
After considerable debugging I can only conclude that the plotly_afterplot event doesn't fire for scatterternary plots.

```
        var plotDiv = document.getElementById(inner_plot_name)
        Plotly.plot(inner_plot_name, data)
        plotDiv.on('plotly_afterplot', function () {
           // never runs this code 
        })
```

Note that this works perfectly for scatter plots, etc.

**Comment by alexcjohnson** [comment]
It's possible that `plotly_afterplot` is being emitted synchronously here - so that it has already fired before you can attach the handler. I was able to confirm that on subsequent replots (such as after zooming) it *is* called, so I suspect that's the case here too. We can look into it and ensure that it gets emitted async, but it might be easier anyway to just use the promise returned by `Plotly.plot`, ie:
```js
Plotly.plot(plotDiv, data)
.then(function() {
    // put code here instead of afterplot
});
```

And if you want that code to get called after zooming or other changes to the plot, you can *also* attach the same function to `plotly_afterplot`.

**Comment by etpinard** [comment]
@dannycodes can I ask what version of plotly.js you're using? I believe this issue what fixed in [`1.23.0`](https://github.com/plotly/plotly.js/releases/tag/v1.23.0). See working example http://codepen.io/etpinard/pen/oxqNvm

**Comment by dannycodes** [comment]
@etpinard Thanks! I was using 1.22.x and updating to 1.25.x solved the problem.
